src/main/java/com/czertainly/api/model/core/settings/Oauth2ClientSettings.java

```diff
@@ -0,0 +1,40 @@
+package com.czertainly.api.model.core.settings;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class Oauth2ClientSettings {
+
+    @Schema(description = "Issuer URI", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String issuerUri;
+
+    @Schema(description = "Client ID", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String clientId;
+
+    @Schema(description = "Client Secret", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String clientSecret;
+
+    @Schema(description = "Authorization Grant Type", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String authorizationGrantType;
+
+    @Schema(description = "Redirect URI", requiredMode = Schema.RequiredMode.REQUIRED)
```

Comment:
This is not configurable, this is managed by the Core. The only meaningful configuration can be Logout Redirect URI